unit_test/blas/Test_Blas1_team_abs.hpp

```diff
@@ -134,7 +134,10 @@ namespace Test {
         expected_result[j] += AT::abs(h_x(i,j)) * AT::abs(h_x(i,j));
     }
 
-    double eps = std::is_same<ScalarA,float>::value?2*1e-5:1e-7;
+//    double eps = std::is_same<ScalarA,float>::value?2*1e-5:1e-7;
+    typename AT::mag_type eps = AT::epsilon()*1000;
```

Comment:
For later: Let's do a bit of numerical analysis here to pick the error bound.  1000*eps is a bit large for small matrices.

Reply:
@mhoemmen 

> Let's do a bit of numerical analysis here to pick the error bound.

Definitely, just a first pass at getting things running with `float` types. The current error checking in some of these tests blows up the error as problem size grows, so this all needs to be revisited.

Reply:
If you look at Test_Blas3_gemm.hpp lines 140 following I actually did this for GEMM. That test should be fine. 
```c++
      double diff_C_average = diff_C/(N*M);
      // Expected Result: Random Walk in the least significant bit (i.e. ~ sqrt(K)*eps
      // eps scales with the total sum and has a factor in it for the accuracy of the operations ->
      // eps = K * 75 * machine_eps * 7
      double diff_C_expected = 1.0*sqrt(K)*K*75*machine_eps*7;
```
Basically you should be able to model this whole thing as a random walk with step size machine epsilon (this is for the additions). Than you add a factor in for the accuracy of the operations involved in this, and then you need the factors in how you aggregate the  errors.

Reply:
This is probably the easiest way.  Another option is to do _a posteriori_ error analysis, using a hand-rolled sequential reimplementation of the algorithm, and a version that replaces sums with absolute values.